crates/rpc/src/handlers/config.rs

```diff
@@ -61,7 +44,16 @@ impl From<Arc<NetworkSpec>> for SpecConfig {
 }
 
 /// Called by `config/spec` to get specification configuration.
-pub async fn get_spec(network_spec: Arc<NetworkSpec>) -> Result<impl Reply, Rejection> {
-    let spec_config = SpecConfig::from(network_spec);
-    Ok(with_status(Data::json(spec_config), StatusCode::OK))
+
+#[get("config/spec")]
+pub async fn get_config_spec() -> impl Responder {
+    let spec_config = SpecConfig::default();
+    HttpResponse::Ok().json(spec_config)
+}
```

Comment:
```suggestion
#[get("config/spec")]
pub async fn get_config_spec() -> impl Responder {
    let spec_config = SpecConfig::default();
    // we shouldn't be using the default of this struct, we should be using the config from the NetworkSpec as done before
    
    HttpResponse::Ok().json(Data::new(spec_config))
}
```
pub async fn get_spec(network_spec: Arc<NetworkSpec>) -> Result<impl Reply, Rejection> {
    let spec_config = SpecConfig::from(network_spec);
    Ok(with_status(Data::json(spec_config), StatusCode::OK))
```